### PR TITLE
update document for tispark-2.2.0 release

### PR DESCRIPTION
**Pull request by marsishandsome**
1. add batch write document to user guide and readme
2. remove tikv config

TODO: need update tidb official website document

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,9 +155,31 @@ spark.sql("select count(*) from lineitem").show
 spark.sql("select ti_version()").show
 ```
 
-## TiDB Data Source API
+## Write Data To TiDB using TiDB Connector
+TiSpark natively supports writing data to TiKV using Spark Data Source API and guarantees ACID.
```
Comment (zhexuany) [inline README.md]: s/TiSpark natively supports writing data to TiKV using Spark Data Source API and guarantees ACID.
Write/TiSpark natively supports writing data to TiKV using Spark Data Source API. ACID is also guaranteed.

Comment (birdstorm) [reply README.md]: I think `and guarantees ACID` is better... You might also change it to `and ACID guarantees`

File: docs/userguide.md [hunk 2]
```diff
@@ -260,9 +225,35 @@ To use TiSpark with Hive:
...
+.mode("append")
+.save()
+```
+
+See [here](./datasource_api_userguide.md) for more details.
+
 ## Load Spark DataFrame into TiDB using JDBC
 
-TiSpark does not provide a direct way to load data into your TiDB cluster, but you can still do it by using JDBC.
+Although TiSpark provides a direct way to load data into your TiDB cluster, you can still do it by using JDBC.
```
Comment (birdstorm) [inline docs/userguide.md]: It is not convincing to use transition here... How about:
```suggestion
While TiSpark provides a direct way to load data into your TiDB cluster, you can also do it by using JDBC.
```

File: docs/userguide_spark2.1.md [hunk 3]
```diff
@@ -277,10 +242,35 @@ val tisparkDF = spark.sql("select * from tispark_table").toDF
 tisparkDF.write.saveAsTable("hive_table") // save table to hive
 spark.sql("select * from hive_table a, tispark_table b where a.col1 = b.col1").show // join table across Hive and Tispark
 ```
+## Load Spark DataFrame into TiDB using TiDB Connector
+TiSpark natively supports writing data to TiKV using Spark Data Source API. ACID is also guaranteed.
```
Comment (birdstorm) [inline docs/userguide_spark2.1.md]:
```suggestion
TiSpark natively supports writing data to TiKV via Spark Data Source API and guarantees ACID.
```
or
```suggestion
TiSpark natively supports writing data to TiKV via Spark Data Source API and ACID guarantees.
```

File: docs/userguide.md [hunk 4]
```diff
@@ -260,9 +225,35 @@ To use TiSpark with Hive:
     spark.sql("select * from hive_table a, tispark_table b where a.col1 = b.col1").show // join table across Hive and Tispark
     ```
 
+## Load Spark DataFrame into TiDB using TiDB Connector
+TiSpark natively supports writing data to TiKV using Spark Data Source API. ACID is also guaranteed.
```
Comment (birdstorm) [inline docs/userguide.md]: same here

File: docs/userguide_spark2.1.md [hunk 5]
```diff
@@ -277,10 +242,35 @@ val tisparkDF = spark.sql("select * from tispark_table").toDF
...
+.mode("append")
+.save()
+```
+
+See [here](./datasource_api_userguide.md) for more details.
 
 ## Load Spark DataFrame into TiDB using JDBC
 
-TiSpark does not provide a direct way to load data into your TiDB cluster, but you can still do it by using JDBC.
+Although TiSpark provides a direct way to load data into your TiDB cluster, you can still do it by using JDBC.
```
Comment (birdstorm) [inline docs/userguide_spark2.1.md]: same here

File: README.md [hunk 6]
```diff
@@ -155,9 +155,31 @@ spark.sql("select count(*) from lineitem").show
 spark.sql("select ti_version()").show
 ```
 
-## TiDB Data Source API
+## Write Data To TiDB using TiDB Connector
+TiSpark natively supports writing data to TiKV using Spark Data Source API. ACID is also guaranteed.
+
+For example:
+
+```scala
```
Comment (ilovesoup) [inline README.md]: Please add comment in code that connection to TiDB is for lock instead of data write.

File: docs/userguide.md (hunk 2 above)
Comment (ilovesoup) [reply docs/userguide.md]: Please put this section a deprecated tag.

File: docs/userguide_spark2.1.md (hunk 3 above)
Comment (ilovesoup) [reply docs/userguide_spark2.1.md]: 2.1 does not support native write right? Why here doc for 2.1 has this?

Comment (marsishandsome) [reply docs/userguide_spark2.1.md]: fixed

File: docs/userguide.md (hunk 2 above)
Comment (marsishandsome) [reply docs/userguide.md]: how to add a [deprecated] tag?

File: README.md (hunk 6 above)
Comment (marsishandsome) [reply README.md]: fixed

**Pull request merged**